lib/solarwinds_apm/opentelemetry/otlp_processor.rb

```diff
@@ -0,0 +1,142 @@
+
+        # return if span is non-entry span
+        return if span.parent_span_id != ::OpenTelemetry::Trace::INVALID_SPAN_ID
+
+        record_request_metrics(span)
+        record_sampling_metrics
+
+        ::OpenTelemetry.meter_provider.metric_readers.each(&:pull)
+        SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] processor on_finish succeed" }
+      rescue StandardError => e
+        SolarWindsAPM.logger.info { "[#{self.class}/#{__method__}] can't flush span to exporter; processor on_finish error: #{e.message}" }
+        ::OpenTelemetry::SDK::Trace::Export::FAILURE
+      end
+
+      private
+
+      # Create two meters for sampling and request count
+      def init_meters
+        {
+          'sw.apm.sampling.metrics' => ::OpenTelemetry.meter_provider.meter('sw.apm.sampling.metrics'),
+          'sw.apm.request.metrics' => ::OpenTelemetry.meter_provider.meter('sw.apm.request.metrics')
+        }
+      end
+
+      def span_attributes(span)
+        span_attrs = { 'sw.transaction' => calculate_lambda_transaction_name(span) }
+        span_attrs.merge!(http_attributes(span))
+        SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] span_attrs: #{span_attrs.inspect}" }
+        span_attrs
+      end
+
+      def meter_attributes(span)
+        meter_attrs = {
+          'sw.is_error' => error?(span) == 1,
+          'sw.transaction' => calculate_lambda_transaction_name(span)
+        }
+
+        meter_attrs.merge!(http_attributes(span))
+        SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] meter_attrs: #{meter_attrs.inspect}" }
+        meter_attrs
+      end
+
+      def calculate_lambda_transaction_name(span)
+        (ENV['SW_APM_TRANSACTION_NAME'] || ENV['AWS_LAMBDA_FUNCTION_NAME'] || span.name || 'unknown').slice(0, 255)
+      end
+
+      def http_attributes(span)
+        return {} unless span_http?(span)
+
+        {
+          'http.status_code' => get_http_status_code(span),
+          'http.method' => span.attributes[HTTP_METHOD]
+        }
+      end
+
+      def init_metrics
+        request_meter = @meters['sw.apm.request.metrics']
+        sampling_meter = @meters['sw.apm.sampling.metrics']
+
+        metrics = {}
+        metrics[:response_time] = request_meter.create_histogram('trace.service.response_time', unit: 'milliseconds')
+        metrics[:tracecount]    = sampling_meter.create_counter('trace.service.tracecount')
+        metrics[:samplecount]   = sampling_meter.create_counter('trace.service.samplecount')
+        metrics[:request_count] = sampling_meter.create_counter('trace.service.request_count')
+        metrics[:toex_count]    = sampling_meter.create_counter('trace.service.tokenbucket_exhaustion_count')
+        metrics[:through_count] = sampling_meter.create_counter('trace.service.through_trace_count')
+        metrics[:tt_count]      = sampling_meter.create_counter('trace.service.triggered_trace_count')
+        metrics
```

Comment:
> just want to extra confirm that's how it is with Ruby too so we don't need to worry about resetting.

Yes, it will keep counting unless metrics reader or exporter  clear the value (e.g. aggregation is delta)